docs/custom-features/db-read-replica.md

```diff
@@ -0,0 +1,9 @@
+# Database Read-Replica
+
+Each deployed environment uses a primary-replica model for the database. Write queries are sent to the primary instance, whilst read queries are balanced between a number of replicas. Replica instances are read-only - any write queries sent to them will return an error. For this reason, database migrations are only run against the primary instance.
+
+To ensure a consistent view of the database, the write instance is always used if there is an open transaction, even for reads.
+
+Generally, a developer shouldn't need worry about this configuration, or explicitly handle it - it is handled automatically by a database router (`cms.core.db_router.ReadReplicaRouter`).
+
+For performance-critical areas, some of Django's built-in queries should be avoided. For example, Django's `get_or_create` will always use the read connection. It may be better to do multiple explicit queries (eg `.get` then `.create`).
```

Comment:
I would change this whole section to either use a [note/important admonition](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) or its own "important" (or "caveats") heading.

And can you please list which queries to avoid?

```markdown
For performance-critical areas, avoid the following Django built-in queries: `get_or_create`, `update_or_create` <insert others here>. For example, `get_or_create` will always use the read connection. It may be better to do multiple explicit queries (eg `.get` then `.create`).
```